js/phemex.js

```diff
@@ -1922,6 +1923,54 @@ module.exports = class phemex extends Exchange {
         return this.parseOrder (data, market);
     }
 
+    async editOrder (id, symbol, type = undefined, side = undefined, amount = undefined, price = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' editOrder() requires a symbol argument');
+        }
+        if (type !== undefined) {
+            throw new ArgumentsRequired (this.id + ' editOrder() type changing is not implemented. Try to cancel & recreate order for that purpose');
+        }
+        if (side !== undefined) {
+            throw new ArgumentsRequired (this.id + ' editOrder() side changing is not implemented. Try to cancel & recreate order for that purpose');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        const clientOrderId = this.safeString2 (params, 'clientOrderId', 'clOrdID');
+        params = this.omit (params, [ 'clientOrderId', 'clOrdID' ]);
+        if (clientOrderId !== undefined) {
+            request['clOrdID'] = clientOrderId;
+        } else {
+            request['orderID'] = id;
+        }
+        const finalPrice = this.safeString (params, 'priceEp');
+        params = this.omit (params, [ 'priceEp' ]);
+        if (finalPrice !== undefined) {
+            request['priceEp'] = this.priceToPrecision (symbol, finalPrice);
```

Comment:
I think, if the user has supplied `priceEp` – it's already formatted to precision, and we should not do `priceToPrecision` here.

Reply:
The same goes to `quantity` below.

Reply:
If we take the above into the account and project this further, then `finalPrice` should not be there at all.
Lines 1948-1954 could be simplified to:
```
if (price !== undefined) {
    request['priceEp'] = this.toEp (price, market);
}
```
The request will be extended with `params` before sending it out anyways, so a user-defined `priceEp` override would take precedence in any case.

Reply:
At, totally right.

Reply:
Lmk what you think now. We also could remove `baseQtyEv` from manual defining approach too, but for the sake of users (as exchange has is erroneously uppercased) I thought support for lowercase would be good too.